### PR TITLE
fix(test): unit test of chain executor

### PR DESCRIPTION
**Pull request by lokax**
## What's changed and what's your intention?

The first message is barrier so that old test code didn't enter the loop correctly

### CODE REVIEW

File: src/stream/src/executor/chain.rs [hunk 1]
```diff
@@ -184,11 +184,19 @@ mod test {
...
+            .next()
+            .await
+            .transpose()
+            .unwrap()
+            .unwrap()
+            .as_barrier()
+            .expect("the first message received by chain must be a barrier");
 
-        let mut count = 0;
+        let mut count = 3;
```
Comment (BugenZhao) [inline src/stream/src/executor/chain.rs:196]: The chain executor will consume the `first` executor at the beginning. So this should be kept `0`?

File: src/stream/src/executor/chain.rs [hunk 2]
```diff
@@ -184,11 +184,19 @@ mod test {
...
         while let Some(Message::Chunk(ck)) = chain.next().await.transpose().unwrap() {
-            count += 1;
             assert_eq!(ck, StreamChunk::from_pretty(&format!("I\n + {count}")));
+            count += 1;
         }
```
Comment (BugenZhao) [inline src/stream/src/executor/chain.rs:200]: Could you please also test the `count` here to check how many chunks we received?

**Review by BugenZhao — COMMENTED** [review]
Thanks for your contribution.

File: src/stream/src/executor/chain.rs (hunk 1 above)
Comment (lokax) [reply src/stream/src/executor/chain.rs:196]: The first barrier mutation is none, so the chain executor just consume the second executor.

Comment (BugenZhao) [reply src/stream/src/executor/chain.rs:189]: Get it. I believe we intended to expect a barrier with mutation in this test... Could you please help to fix this?

**Review by BugenZhao — APPROVED** [review]
LGTM!

**Pull request merged**